File: WebAPI/WebAPI.Models/Models/WorkSpace.cs
using System;

namespace WebAPI.Models.Models
{
    public class WorkSpace
    {
        public Guid WorkSpaceId { get; set; }
        
        public string WorkSpaceName { get; set; }
        
        public string WorkSpaceDescription { get; set; }
        
        public DateTime CreationDate { get; set; }
    }
}

File: web/src/types/workspace.ts
import { IProjectSimpleModel } from './project';
import { ITeamSimpleModel } from './team';

export interface IWorkSpace {
    workSpaceId?: string;
    workSpaceName: string;
    workSpaceDescription: string;
    creationDate?: Date;
}

export interface IWorkSpacePageProject extends IProjectSimpleModel {
    teams: ITeamSimpleModel[];
}

export interface IWorkSpacePage {
    workSpace: IWorkSpace;
    projects: IWorkSpacePageProject[];
}

export interface ISearchResults {
    projects: IProjectSimpleModel[];
    teams: ITeamSimpleModel[];
}
